STYLEGUIDE.md

```diff
@@ -0,0 +1,288 @@
+
+static u8 sMyStaticVariable = 0;
+```
+
+Macros and constants should use `CAPS_WITH_UNDERSCORES`.
+
+```c
+#define MAX_LEVEL 100
+
+enum
+{
+    COLOR_RED,
+    COLOR_BLUE,
+    COLOR_GREEN,
+};
+
+#define ADD_FIVE(x) ((x) + 5)
+```
+
+## Coding Style
+
+### Comments
+
+When describing a system/component in-depth, use block comment syntax.
+
+```c
+/*
+ * This is an in-depth description of the save block format. Its format is as follows:
+ *
+ * Sectors  0 - 13: Save Slot 1
+ * Sectors 14 - 27: Save Slot 2
+ * ...
+ */
+```
+
+When briefly describing a function or block of code, use a single-line comments
+placed on its own line. 
+There should be a single space directly to the right of `//`.
+
+```c
+// This is supplemental information for the function. If there is a bunch of info, it should
+// carry on to the next line.
+void ProcessSingleTask(void)
+{
+   // Short comment describing some noteworthy aspect of the code immediately following.
+   ...
+   // Comments should be capitalized and end in a period.
+}
+```
+
+When tagging a data structure that corresponds to an `enum` or some noteworthy
+value, place the comment on the same line as the code.
+```c
+const u8 gPlantlikeMons[] =
+{
+    FALSE, // SPECIES_BULBASAUR
+    FALSE, // SPECIES_IVYSAUR
+    TRUE,  // SPECIES_VENUSAUR
+    FALSE, // SPECIES_CHARMANDER
+    ...
+};
+```
+
+### Whitespace
+
+All `.c` and `.h` files should use 4 spaces--not tabs.
+Assembler files (`.s)` use tabs.
+
+### Operators
+
+Assignments and comparison operators should have one space on both sides of `=`.
+
+```c
+int i = 0; // correct
+int i=0;   // incorrect
+
+a > b // correct
+a>b   // incorrect
+```
+
+The incrementor and decrementor operators should NOT have a space.
+
+```c
+i++;  // correct
+i ++; // incorrect
+```
+
+A control statement should have a space between them and their expressions, and the opening bracket should be on the next line.
+
+```c
+for (...)
+{
+    // correct
+}
+
+for(...) {
+    // incorrect
+}
+```
+
+A `switch` statement's cases should left-align with the `switch`'s block.
+
+```c
+switch (foo)
+{
+case 0: // correct
+    ...
+    break;
+}
+
+switch (foo)
+{
+    case 0: // incorrect
+        ...
+        break;
+}
+```
+
+A single empty line should follow a block.
+
+```c
+int MyFunction(int bar)
+{
+    int foo = 0;
+    if (bar)
+        foo++;
+
+    return foo; // incorrect
+}
+
+int MyFunction(int bar)
+{
+    int foo = 0;
+    if (bar)
+        foo++;
+    return foo; // incorrect
+}
+```
+
+A chain of `if-else` statements in which any block is more than one line of
+code should use braces. If all blocks are single-line, then no braces are necessary.
+
+```c
+if (foo) // correct
+{
+    return 1;
+}
+else
+{
+    MyFunction();
+    return 0;
+}
+
+if (foo) // incorrect
+    return 1;
+else
+{
+    MyFunction();
+    return 0;
+}
+```
+
+### Control Structures
+
+When comparing whether or not a value equals `0`, don't be explicit unless the
+situation calls for it.
```

Comment:
Probably worth mentioning that we do `== TRUE`/`== FALSE` _all over the place_. It's bizarre to draw a distinction between `0` and `FALSE` in this way; it's exactly backward of how you'd expect drawing a distinction to shake out, e.g. we're saying to only write `==` in situations where languages like Java with a `boolean` type would not require it, and omit `==` where it would require it.

```
$ git grep '== \(TRUE\|FALSE\)' pret/master | wc -l
1038
```